layers/containers/custom_containers.h

```diff
@@ -26,7 +26,7 @@
 #include <utility>
 
 #ifdef USE_CUSTOM_HASH_MAP
-#include "robin_hood.h"
+#include "external/parallel_hashmap/phmap.h"
```

Comment:
you shouldn't need `external` here, we already have the folder included as a target include already

```suggestion
#include "parallel_hashmap/phmap.h"
```

Reply:
I did that for some test complaining about not finding `parallel_hashmap/phmap.h`

Reply:
I hope no test needs `custom_container.h`

if they do, you should update the `target_include_directories` in the CMake like you did for the gpuav-spirv cmake

Reply:
I will fix the test

Reply:
> C:\Users\Arno\lunarg\Vulkan-ValidationLayers\layers\containers\custom_containers.h(29,10): error C1083: Cannot open include file: 'parallel_hashmap/phmap.h': No such file or directory [C:\Users\Arno\
lunarg\Vulkan-ValidationLayers\build\debug\tests\layers\VkLayer_device_profile_api.vcxproj]
  (compiling source file '../../../../tests/layers/device_profile_api.cpp')

Reply:
this seems like bad header includes, I see no use of `vvl::` inside `device_profile_api.cpp`

Reply:
`device_profile_api.cpp` includes `vk_dispatch_table_helper.h` that includes `vk_extension_helper.h` that includes `custom_containers.h`

Reply:
but then I just need to add this:
```
target_include_directories(VkLayer_device_profile_api PRIVATE .
    ${VVL_SOURCE_DIR}/layers/external
)
```

Reply:
No, just move the header to the source and forward declare instead of hacking this with CMake

```diff
diff --git a/layers/vulkan/generated/vk_dispatch_table_helper.cpp b/layers/vulkan/generated/vk_dispatch_table_helper.cpp
index c2146c620..625846c12 100644
--- a/layers/vulkan/generated/vk_dispatch_table_helper.cpp
+++ b/layers/vulkan/generated/vk_dispatch_table_helper.cpp
@@ -1957,6 +1957,7 @@ const auto& GetApiExtensionMap() {
     return api_extension_map;
 }
 
+#include "vk_extension_helper.h"
 // Using the above code-generated map of APINames-to-parent extension names, this function will:
 //   o  Determine if the API has an associated extension
 //   o  If it does, determine if that extension name is present in the passed-in set of device or instance enabled_ext_names
diff --git a/layers/vulkan/generated/vk_dispatch_table_helper.h b/layers/vulkan/generated/vk_dispatch_table_helper.h
index 14d671e4a..e57140aac 100644
--- a/layers/vulkan/generated/vk_dispatch_table_helper.h
+++ b/layers/vulkan/generated/vk_dispatch_table_helper.h
@@ -30,7 +30,8 @@
 #include <cstring>
 #include <string>
 #include "vk_layer_dispatch_table.h"
-#include "vk_extension_helper.h"
+
+struct DeviceExtensions;
 
 // Using the above code-generated map of APINames-to-parent extension names, this function will:
 //   o  Determine if the API has an associated extension
```

Reply:
`vk_extension_helper.h` is being included in tests, with no way to remove references to `custom_containers.h` since we need vvl maps here:
![image](https://github.com/user-attachments/assets/042628a3-366b-4cd7-8b50-8d56ac995b1e)

so we need `custom_containers.h` in tests

Reply:
Still I made the change you mentioned to `vk_dispatch_table_helper`, one less include is always nice

Reply:
... this is 100% getting outside the scope of this PR, can rry and cleanup after, I already spotted many random includes in the test code now... so sure, signed off for this part of the change, thanks for making the small change while at it